flink-python/src/main/java/org/apache/flink/streaming/api/operators/python/TimestampsAndWatermarksOperator.java

```diff
@@ -0,0 +1,227 @@
+import org.apache.flink.streaming.api.operators.ChainingStrategy;
+import org.apache.flink.streaming.api.operators.Output;
+import org.apache.flink.streaming.api.runners.python.beam.BeamDataStreamPythonFunctionRunner;
+import org.apache.flink.streaming.api.utils.PythonOperatorUtils;
+import org.apache.flink.streaming.api.utils.PythonTypeUtils;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.apache.flink.streaming.runtime.streamstatus.StreamStatus;
+import org.apache.flink.streaming.runtime.streamstatus.StreamStatusMaintainer;
+import org.apache.flink.streaming.runtime.tasks.ProcessingTimeCallback;
+import org.apache.flink.types.Row;
+
+import java.util.Collections;
+
+/**
+ * A stream operator that may do one or both of the following: extract timestamps from
+ * events and generate watermarks by user specify TimestampAssigner and WatermarkStrategy.
+ *
+ * <p>These two responsibilities run in the same operator rather than in two different ones,
+ * because the implementation of the timestamp assigner and the watermark generator is
+ * frequently in the same class (and should be run in the same instance), even though the
+ * separate interfaces support the use of different classes.
+ *
+ * @param <IN> The type of the input elements
+ */
+public class TimestampsAndWatermarksOperator<IN> extends StatelessOneInputPythonFunctionOperator<IN, IN>
+	implements ProcessingTimeCallback {
+
+	private static final long serialVersionUID = 1L;
+
+	private final WatermarkStrategy<IN> watermarkStrategy;
+
+	private final TypeInformation runnerInputTypeInfo;
+
+	private final TypeInformation runnerOutputTypeInfo;
+
+	private transient TypeSerializer runnerInputSerializer;
+
+	private transient TypeSerializer runnerOutputSerializer;
+
+	private transient WatermarkGenerator<IN> watermarkGenerator;
+
+	private transient WatermarkOutput watermarkOutput;
+
+	private transient long watermarkInterval;
+
+	private transient Row resuableInput;
+
+	private transient StreamRecord<IN> reusableStreamRecord;
+
+	public TimestampsAndWatermarksOperator(
+		Configuration config,
+		TypeInformation<IN> inputTypeInfo,
+		DataStreamPythonFunctionInfo pythonFunctionInfo,
+		WatermarkStrategy<IN> watermarkStrategy) {
+		super(config, inputTypeInfo, inputTypeInfo, pythonFunctionInfo);
+		this.watermarkStrategy = watermarkStrategy;
+		this.chainingStrategy = ChainingStrategy.ALWAYS;
+		this.runnerInputTypeInfo = Types.ROW(Types.LONG, inputTypeInfo);
+		this.runnerOutputTypeInfo = Types.ROW(Types.LONG, inputTypeInfo);
+	}
+
+	@Override
+	public void open() throws Exception {
+		super.open();
+		runnerInputSerializer = PythonTypeUtils.TypeInfoToSerializerConverter
+			.typeInfoSerializerConverter(runnerInputTypeInfo);
+		runnerOutputSerializer = PythonTypeUtils.TypeInfoToSerializerConverter
+			.typeInfoSerializerConverter(runnerOutputTypeInfo);
+		resuableInput = new Row(2);
+		this.reusableStreamRecord = new StreamRecord<>(null);
```

Comment:
```suggestion
		reusableStreamRecord = new StreamRecord<>(null);
```
Make the code style consistent.